enforcer/pucontext/pucontext.go

```diff
@@ -13,25 +13,132 @@ import (
 
 // PUContext holds data indexed by the PU ID
 type PUContext struct {
-	ID                string
-	ManagementID      string
-	Identity          *policy.TagStore
-	Annotations       *policy.TagStore
-	AcceptTxtRules    *lookup.PolicyDB
-	RejectTxtRules    *lookup.PolicyDB
-	AcceptRcvRules    *lookup.PolicyDB
-	RejectRcvRules    *lookup.PolicyDB
-	ApplicationACLs   *acls.ACLCache
-	NetworkACLS       *acls.ACLCache
-	ExternalIPCache   cache.DataStore
-	Extension         interface{}
-	IP                string
-	Mark              string
-	ProxyPort         string
-	Ports             []string
-	PUType            constants.PUType
-	SynToken          []byte
-	SynServiceContext []byte
-	SynExpiration     time.Time
+	ID                    string
+	ManagementID          string
+	Identity              *policy.TagStore
+	Annotations           *policy.TagStore
+	rejectTxtRules        *lookup.PolicyDB
+	observeRejectTxtRules *lookup.PolicyDB
+	acceptTxtRules        *lookup.PolicyDB
+	observeAcceptTxtRules *lookup.PolicyDB
+	rejectRcvRules        *lookup.PolicyDB
+	observeRejectRcvRules *lookup.PolicyDB
+	acceptRcvRules        *lookup.PolicyDB
+	observeAcceptRcvRules *lookup.PolicyDB
+	ApplicationACLs       *acls.ACLCache
+	NetworkACLS           *acls.ACLCache
+	ExternalIPCache       cache.DataStore
+	Extension             interface{}
+	IP                    string
+	Mark                  string
+	ProxyPort             string
+	Ports                 []string
+	PUType                constants.PUType
+	SynToken              []byte
+	SynServiceContext     []byte
+	SynExpiration         time.Time
 	sync.Mutex
 }
+
+// createRuleDBs creates the database of rules from the policy
+func (p *PUContext) createRuleDBs(policyRules policy.TagSelectorList) (*lookup.PolicyDB, *lookup.PolicyDB, *lookup.PolicyDB, *lookup.PolicyDB) {
+
+	rejectRules := lookup.NewPolicyDB()
+	rejectObserveRules := lookup.NewPolicyDB()
+	acceptRules := lookup.NewPolicyDB()
+	acceptObserveRules := lookup.NewPolicyDB()
+
+	for _, rule := range policyRules {
+		if rule.Policy.Action.Accepted() {
+			if rule.Policy.Action.Observed() {
+				acceptObserveRules.AddPolicy(rule)
+			} else {
+				acceptRules.AddPolicy(rule)
+			}
+		} else if rule.Policy.Action.Rejected() {
+			if rule.Policy.Action.Observed() {
+				rejectObserveRules.AddPolicy(rule)
+			} else {
+				rejectRules.AddPolicy(rule)
+			}
+		} else {
+			continue
```

Comment:
if we reach here, should it be a panic? how can a policy be neither accepted nor rejected?